### ISSUE TITLE
Possible Typo in LayeredBxDF::PDF?

### ISSUE THREAD

**Issue by anooprthomas** [mmp/pbrt-v4#214]
in bxdfs,h, LayeredBxDF::PDF (line 866)

```
                uc = r();
                u = Point2f(r(), r());
                pstd::optional<BSDFSample> wis = tiInterface.Sample_f(wi, uc, u, !mode);
-->           if (!wis || !wos->f || wos->pdf == 0 || wos->wi.z == 0 ||
                    wis->IsReflection())
                    continue;
```

Should that condition be the following instead? :
`if (!wis || !wis->f || wis->pdf == 0 || wis->wi.z == 0 ||`

**Comment by mmp** [comment]
Yikes! Good find--you're correct. Fixed in 981e2bf72a4786182f4b3a0478bb626bec87dd91. Thank you!!

**Comment by anooprthomas** [comment]
you're welcome :). glad to be of some help :). I'm currently using this repo as reference for my GPU + RTX based implementation. Will keep an eye out for any more issues.
